app_confetti/config.py

```diff
@@ -1,21 +1,26 @@
+import dataclasses
+
+from app_confetti import util
+
+
+@dataclasses.dataclass(frozen=True)
 class BaseConfig:
     """Configuration class.
 
-    The `environ_config` library allows for its attributes to be retrieved from
-    environment variables, the prefix for this class and the attribute name
-    defines the variable name to be used, e.g. PREFIX_SENTRY_DSN for SENTRY_DSN.
+    The `env` util function allows for attributes to be retrieved from environment variables.
 
-    This class is generally used in settings.__init__.py to instantiate an object from
-    the environment.
-
-    Inheriting classes must implement:
-        cls.LOGGING_LEVEL
-        cls.SENTRY_DSN
-        cls.ENV
+    Inheriting classes can override:
+        logging_level: str
+        sentry_dsn: str
+        env: str
     """
 
+    logging_level: str = util.env("LOGGING_LEVEL:INFO")
+    sentry_dsn: int = util.env("SENTRY_DSN:None", util.str_to_literal)
+    env: int = util.env("ENV:None", util.str_to_literal)
+
     @property
```

Comment:
Try to avoid these computed properties, keep the Config _dumb and flat_. Building configuration should be the responsibility of the application, not the Config object (https://en.wikipedia.org/wiki/Single-responsibility_principle). A utility function to build this mapping would be more suitable.

This also means you can build config using different methods that might not comply with `dictConfig`, for example using https://github.com/Delgan/loguru. The developer wants to be able to look at the Config Class in the project and see _all_ the values instead of having to dive into its parent(s) to find ones mapped in via inheritance.

I think we should tear out this base config and just have a self contained Config Class in the projects with the vars it needs; if there is a Config that _doesn't_ need a sentry_dsn, it shouldn't be looking for one.

Reply:
Yeah the base config belongs in CLT repos, as they all use the same logging, it doesn't belong in the root lib.